src/main/java/org/broadinstitute/hellbender/tools/walkers/mutect/filtering/HardAlleleFilter.java

```diff
@@ -0,0 +1,29 @@
+package org.broadinstitute.hellbender.tools.walkers.mutect.filtering;
+
+import htsjdk.variant.variantcontext.Allele;
+import htsjdk.variant.variantcontext.VariantContext;
+import org.apache.commons.collections4.map.ListOrderedMap;
+import org.broadinstitute.hellbender.engine.ReferenceContext;
+
+import java.util.*;
+
+public abstract class HardAlleleFilter<T> extends Mutect2AlleleFilter<T> {
+    public void calculateErrorProbabilityForAlleles(ListOrderedMap<Allele, Double> results, final ListOrderedMap<Allele, List<T>> dataByAllele, final VariantContext vc, final Mutect2FilteringEngine filteringEngine, ReferenceContext referenceContext) {
```

Comment:
I appreciate the use of ordered maps, but why use a `ListOrderedMap` instead of the more standard `LinkedHashmap`?